data/trie/trieStorageManager.go

```diff
@@ -117,6 +119,30 @@ func (tsm *trieStorageManager) Database() data.DBWriteCacher {
 	return tsm.db
 }
 
+// EnterSnapshotMode sets the snapshot mode on
+func (tsm *trieStorageManager) EnterSnapshotMode() {
+	tsm.storageOperationMutex.Lock()
+	defer tsm.storageOperationMutex.Unlock()
+
+	tsm.snapshotInProgress++
+}
+
+// ExitSnapshotMode sets the snapshot mode off
+func (tsm *trieStorageManager) ExitSnapshotMode() {
+	tsm.storageOperationMutex.Lock()
+	defer tsm.storageOperationMutex.Unlock()
+
+	if tsm.snapshotInProgress > 0 {
```

Comment:
I would add a log error in case snapshotInProgress < 1

Reply:
done.